### DISCUSSION TITLE
impossible to connect

### DISCUSSION THREAD

**Discussion by zunnnn** [wlcrs/huawei_solar#144] [Setting up the integration]
hi, i have the dongle correctly configured and i can see the wired lan and the wireless lan corretly connected to my lan:
![image](https://user-images.githubusercontent.com/60466697/194698066-74ad2834-aa9a-40d6-bc36-c8428c652a3c.png)
221 is wifi
222 is wired
so 222 dosn't have any port open, but the 221 have the 502 like the image
![image](https://user-images.githubusercontent.com/60466697/194698108-24804410-cc75-4712-87d9-bd846bed337a.png)
i would like to use the option B
![image](https://user-images.githubusercontent.com/60466697/194698177-62ae1741-0a6f-48ac-b504-539121de9ee4.png)
but i have an error when i try to setup the integration on HA, couldn't connect
i can connect with qmodmaster to 192.168.1.221:508 in TCP, but i can't get any data, if i try to switch to rtu i get an error
![image](https://user-images.githubusercontent.com/60466697/194698322-bc21a132-c7f3-4859-83e2-142734b07a88.png)
![image](https://user-images.githubusercontent.com/60466697/194698331-97bc8a44-4ccf-43cf-80f7-470027f094cf.png)

any ideas?

**Comment by bengtner** [comment]
Make sure you have the right firmware in place. Check both the inverter and the dongle using the Fusional Solar app connected to the inverter over the inverter WLAN and the dongle via the dongle. You choose which connection to use inside the app.

Once you are connected to the inverter, make sure you the TCP over Modbus option is enabled.

**Reply by zunnnn** [reply]
> Make sure you have the right firmware in place. Check both the inverter and the dongle using the Fusional Solar app connected to the inverter over the inverter WLAN and the dongle via the dongle. You choose which connection to use inside the app.
> 
> Once you are connected to the inverter, make sure you the TCP over Modbus option is enabled.

i think i found the problem:
![image](https://user-images.githubusercontent.com/60466697/194716583-52b4f668-fb4c-48d1-9d89-9049c15f3757.png)
old firmware
how can i upgrade?
update:
checked from inverter AP and mobile app in "installation mode" i got this:
![image](https://user-images.githubusercontent.com/60466697/194716856-bf2a4588-323b-410b-9bd2-6ea817ce79c2.png)

i have got only inverter, battery, optimizer, and mbus witch have a different version number from dongle.

**Reply by bengtner** [reply]
You use the Fusion Solar app to upgrade. But the problem  is the late(st) versions of software aren't generally available. You have to issue a request to Huawei support to get the right versions. I don't know why they don't distribute those version. Not even the installer has the latest versions.

**Comment by wlcrs** [comment]
Have you checked the slave id/unit id? What values have you tried?

**Reply by zunnnn** [reply]
> Have you checked the slave id/unit id? What values have you tried?

i have set n° 1 and use it in configuration (also tried 0)

**Comment by alexualbu** [comment]
did you manage to make it work with newer firmware?
